python/ray/util/sgd/v2/trainer.py

```diff
@@ -55,14 +55,19 @@ class Trainer:
         logdir (Optional[str]): Path to the file directory where logs
             should be persisted. If this is not specified, one will be
             generated.
+         max_retries (Optional[int]): Number of retries when Ray actors fail.
+            Defaults to 3. Set to -1 for unlimited retries or 0 to disable.
```

Comment:
Can we change this behavior so that:
1. `None` means unlimited retries (i.e. no max retries)
2. Negative numbers are disallowed.

Reply:
I think -1 makes more sense in our context (used as convention for Core, Tune too)

Reply:
Fair enough, can we remove the `Optional` then?

Reply:
Done